IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -240,14 +240,18 @@ - (BOOL)saveAccount:(nonnull MSIDAccountCacheItem *)account
 
     MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, context, @"(Default cache) Saving account %@", MSID_PII_LOG_MASKABLE(account));
 
-    MSIDDefaultAccountCacheKey *key = [[MSIDDefaultAccountCacheKey alloc] initWithHomeAccountId:account.homeAccountId
-                                                                                    environment:account.environment
-                                                                                          realm:account.realm
-                                                                                           type:account.accountType];
-
+    MSIDDefaultAccountCacheKey *key = (MSIDDefaultAccountCacheKey *)[account generateCacheKey];
+    
+    if (![key isKindOfClass:[MSIDDefaultAccountCacheKey class]])
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelWarning,context, @"Save account called for key which is not of type {MSIDDefaultAccountCacheKey}.");
```

Comment:
nit: space missing for `MSIDLogLevelWarning,context,`

Reply:
Addressed. Remove this line of code as I removed the need to cast the key.